core/wait/wait.go

```diff
@@ -79,3 +92,22 @@ func (w *Handler) WaitWithContext(ctx context.Context) (res interface{}, err err
 		}
 	}
 }
+
+func handleError(retryTempErrorCounter, retryLimitTempErr int, err error) (int, error) {
```

Comment:
Maybe we could add `(w *Handler)` as a receiver here so we don't have to pass down `retryLimitTempErr`